mrp_subcontracting_purchase/models/stock_rule.py

```diff
@@ -0,0 +1,207 @@
+from collections import defaultdict
+from itertools import groupby
+
+from dateutil.relativedelta import relativedelta
+
+from odoo import SUPERUSER_ID, _, api, fields, models
+from odoo.tools import float_compare
+
+from odoo.addons.stock.models.stock_rule import ProcurementException
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.rule"
+
+    @api.model
+    def _run_buy(self, procurements):
+        """Launching a purchase group with required/custom
+        fields generated by a sales order line"""
+        procurements_by_po_domain = defaultdict(list)
+        errors = []
+        message = _(
+            """There is no matching vendor price to generate the purchase order for
+                product %s (no vendor defined,  minimum quantity not reached,
+                dates not valid, ...).
+                Go on the product form and complete the list of vendors."""
+        )
+        for procurement, rule in procurements:
+            supplier = self._get_supplier(procurement)
+            if not supplier:
+                errors.append(
+                    (procurement, message % (procurement.product_id.display_name))
+                )
+            partner = supplier.name
+            # we put `supplier_info` in values for extensibility purposes
+            procurement.values.update(
+                {"supplier": supplier, "propagate_cancel": rule.propagate_cancel}
+            )
+            domain = rule._make_po_get_domain(
+                procurement.company_id, procurement.values, partner
+            )
+            procurements_by_po_domain[domain].append((procurement, rule))
+        if errors:
+            raise ProcurementException(errors)
+        self._create_po_not_exist(procurements_by_po_domain)
+
+    def _prepare_purchase_order(self, company_id, origins, values):
+        """Returns prepared data for create PO"""
+        if (
+            "partner_id" not in values[0]
```

Comment:
`value` may be empty? If `value` will be empty than you need check this before this condition.

Reply:
value is checking [here](https://github.com/OCA/manufacture/pull/1082/files/adf1f14f9e0686498b279baa8cd7fd1b899aa259#diff-b8f74f699c3783e44e9e4347460bac8b9c835865069caa66f1fdce52d8f9385bR182-R188)